agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -132,19 +134,16 @@ private static String getFriendlyExceptionBanner(String url) {
     return "ApplicationInsights Java Agent failed to send telemetry data.";
   }
 
-  private static String getCipherFriendlyExceptionAction(String url, List<String> missingCiphers) {
+  private static String getCipherFriendlyExceptionAction(String url) {
     StringBuilder actionBuilder = new StringBuilder();
-    actionBuilder
-        .append("The following are the cipher suites missing from Java runtime: ")
-        .append("\n");
-    for (String missingCipher : missingCiphers) {
-      actionBuilder.append(missingCipher).append("\n");
-    }
     actionBuilder
         .append(
-            "Please add at least one of the missing cipher suites that are expected from the target endpoint:"
+            "The Application Insights Java agent detects that you do not have any of the following cipher suites that are supported by the endpoint it connects to: "
```

Comment:
let's log the cipher suites that were found as well

Reply:
By adding this, I think the friendly exception will be too verbose. I was hoping users can get that information from the code snippet present in trouble shooting docs.

Reply:
can you show an example of what it will look like? i think it may be helpful to know, e.g., if there are no cipher suites available, or possibly some really old ones

Reply:
Ciphers From JVM:
*	TLS_AES_256_GCM_SHA384
*	TLS_AES_128_GCM_SHA256
*	TLS_CHACHA20_POLY1305_SHA256
*	TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
*	TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
*	TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
*	TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
*	TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
*	TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
*	TLS_DHE_RSA_WITH_AES_256_GCM_SHA384
*	TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256
*	TLS_DHE_DSS_WITH_AES_256_GCM_SHA384
*	TLS_DHE_RSA_WITH_AES_128_GCM_SHA256
*	TLS_DHE_DSS_WITH_AES_128_GCM_SHA256
*	TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
*	TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
*	TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
*	TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
*	TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
*	TLS_DHE_DSS_WITH_AES_256_CBC_SHA256
*	TLS_DHE_RSA_WITH_AES_128_CBC_SHA256
*	TLS_DHE_DSS_WITH_AES_128_CBC_SHA256
*	TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384
*	TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384
*	TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256
*	TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256
*	TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384
*	TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384
*	TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256
*	TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256
*	TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
*	TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
*	TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
*	TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
*	TLS_DHE_RSA_WITH_AES_256_CBC_SHA
*	TLS_DHE_DSS_WITH_AES_256_CBC_SHA
*	TLS_DHE_RSA_WITH_AES_128_CBC_SHA
*	TLS_DHE_DSS_WITH_AES_128_CBC_SHA
*	TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA
*	TLS_ECDH_RSA_WITH_AES_256_CBC_SHA
*	TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA
*	TLS_ECDH_RSA_WITH_AES_128_CBC_SHA
*	TLS_RSA_WITH_AES_256_GCM_SHA384
*	TLS_RSA_WITH_AES_128_GCM_SHA256
*	TLS_RSA_WITH_AES_256_CBC_SHA256
*	TLS_RSA_WITH_AES_128_CBC_SHA256
*	TLS_RSA_WITH_AES_256_CBC_SHA
*	TLS_RSA_WITH_AES_128_CBC_SHA
*	TLS_EMPTY_RENEGOTIATION_INFO_SCSV

Reply:
but we wouldn't log in this case because it has at least one of the expected ciphers, what about in the jlink repro?

Reply:
oh my bad. Here is a sample message with complete list from jlink repro:
*************************
ApplicationInsights Java Agent failed to connect to Live metric end point.
*************************

Description:
Probable root cause may be : missing cipher suites which are expected by the requested target.

Action:
The Application Insights Java agent detects that you do not have any of the following cipher suites that are supported by the endpoint it connects to: https://rt.services.visualstudio.com/QuickPulseService.svc
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256

List of existing ciphers from JVM:
TLS_AES_256_GCM_SHA384
TLS_AES_128_GCM_SHA256
TLS_CHACHA20_POLY1305_SHA256
TLS_DHE_RSA_WITH_AES_256_GCM_SHA384
TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256
TLS_DHE_DSS_WITH_AES_256_GCM_SHA384
TLS_DHE_RSA_WITH_AES_128_GCM_SHA256
TLS_DHE_DSS_WITH_AES_128_GCM_SHA256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
TLS_DHE_DSS_WITH_AES_256_CBC_SHA256
TLS_DHE_RSA_WITH_AES_128_CBC_SHA256
TLS_DHE_DSS_WITH_AES_128_CBC_SHA256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA
TLS_DHE_DSS_WITH_AES_256_CBC_SHA
TLS_DHE_RSA_WITH_AES_128_CBC_SHA
TLS_DHE_DSS_WITH_AES_128_CBC_SHA
TLS_RSA_WITH_AES_256_GCM_SHA384
TLS_RSA_WITH_AES_128_GCM_SHA256
TLS_RSA_WITH_AES_256_CBC_SHA256
TLS_RSA_WITH_AES_128_CBC_SHA256
TLS_RSA_WITH_AES_256_CBC_SHA
TLS_RSA_WITH_AES_128_CBC_SHA
TLS_EMPTY_RENEGOTIATION_INFO_SCSV
Learn more about troubleshooting this network issue related to cipher suites here: https://go.microsoft.com/fwlink/?linkid=2185426

Note:
This message is only logged the first time it occurs after startup.